examples/react-native/storybook/stories/Authenticator.stories.tsx

```diff
@@ -0,0 +1,181 @@
+    placeholder: 'Phone',
+    type: 'phone' as const,
+  },
+  username: {
+    name: 'username',
+    label: 'Username',
+    placeholder: 'Username',
+    type: 'default' as const,
+  },
+};
+
+type Fields = keyof typeof componentFields;
+
+const getComponentFields = (fields: Fields[]) => {
+  return {
+    fields: fields.map((field) => componentFields[field]),
+  };
+};
+
+const getComponentSlots = (subcomponent: any) => {
+  return {
+    Footer: subcomponent.Footer,
+    FormFields: subcomponent.FormFields,
+    Header: subcomponent.Header,
+  };
+};
+
+const sharedProps = {
+  error: null as unknown as string,
+  handleBlur: noop,
+  handleChange: noop,
+  handleSubmit: (values: any) => {
+    console.log('Values', values);
+  },
+  isPending: false,
+};
+
+const confirmResetPasswordProps = {
+  ...sharedProps,
+  ...getComponentFields(['code', 'newPassword', 'confirmPassword']),
+  ...getComponentSlots(Authenticator.ConfirmResetPassword),
+  resendCode: noop,
+};
+
+const confirmSignInProps = {
+  ...sharedProps,
+  ...getComponentFields(['code']),
+  ...getComponentSlots(Authenticator.ConfirmSignIn),
+  challengeName: 'SMS_MFA' as AuthChallengeName,
+  toSignIn: noop,
+};
+
+const confirmSignUpProps = {
+  ...sharedProps,
+  ...getComponentFields(['code']),
+  ...getComponentSlots(Authenticator.ConfirmSignUp),
+  codeDeliveryDetails: {
+    AttributeName: 'email',
+    DeliveryMedium: 'EMAIL',
+    Destination: 'a***@e***.com',
+  } as CodeDeliveryDetails,
+  resendCode: noop,
+};
+
+const confirmVerifyUserProps = {
+  ...sharedProps,
+  ...getComponentFields(['username']),
+  ...getComponentSlots(Authenticator.ConfirmResetPassword),
+  skipVerification: noop,
+};
+
+const forceNewPasswordProps = {
+  ...sharedProps,
+  ...getComponentFields(['newPassword', 'confirmPassword']),
+  ...getComponentSlots(Authenticator.ForceNewPassword),
+  toSignIn: noop,
+};
+
+const resetPasswordProps = {
+  ...sharedProps,
+  ...getComponentFields(['username']),
+  ...getComponentSlots(Authenticator.ResetPassword),
+  toSignIn: noop,
+};
+
+const setupTOTPProps = {
+  ...sharedProps,
+  ...getComponentFields(['code']),
+  ...getComponentSlots(Authenticator.SetupTOTP),
+  getTotpSecretCode: noop as GetTotpSecretCode,
+  toSignIn: noop,
+};
+
+const signInProps = {
+  ...sharedProps,
+  ...getComponentFields(['username', 'password']),
+  ...getComponentSlots(Authenticator.SignIn),
+  toFederatedSignIn: noop,
+  toResetPassword: noop,
+  toSignUp: noop,
+};
+
+const signUpProps = {
+  ...sharedProps,
+  ...getComponentFields(['username', 'password', 'confirmPassword', 'phone']),
+  ...getComponentSlots(Authenticator.SignUp),
+  toFederatedSignIn: noop,
+  toSignIn: noop,
+};
+
+const verifyUserProps = {
+  ...sharedProps,
+  ...getComponentSlots(Authenticator.VerifyUser),
+  fields: [],
+  skipVerification: noop,
+};
+
+storiesOf('Authenticator', module)
+  .addDecorator((Story: any) => (
+    <InnerContainer>
+      <Story />
+    </InnerContainer>
+  ))
```

Comment:
We need the `DefaultContainer` too right?

Reply:
I don't think we need to because we're already using the `Screen` component as a decorator for each of our stories, and it has the same styles as `Default Container`:

```
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  ```

Reply:
What if the `DefaultContainer` changes?

Reply:
Hmm. Maybe we can replace the `Screen` component with `DefaultContainer`?